### PR TITLE
initial push, WIP

### PR DESCRIPTION
**Pull request by rossdavidh**
Not actually done yet,  just PR'ing so you can see WIP more easily.

### CODE REVIEW

File: plaster/run/call_bag.py [hunk 1]
```diff
@@ -313,123 +312,33 @@ def pr_curve_old(self, pep_iz_subset=None, n_steps=50):
...
+        Similar format to pr_curve_pep, but is PR curve for proteins, not peptides
         Returns a tuple of arrays; each row of the arrays is an increasing score threshold. The arrays are:
             * precision, recall, score_thresh, area_under_curve
         """
 
         # Obtain a reverse sorted calls: true, pred, score
-        true = self.df["true_pep_iz"].values
-        pred = self.df["pred_pep_iz"].values
+        true = self.true_peps__pros()["pro_i"].values
+        pred = self.pred_peps__pros()["pro_i"].values
```
Comment (zack-erisyon) [inline plaster/run/call_bag.py:334]: Better form is to use .pro_i like...
self.true_peps__pros().pro_i.values

File: plaster/run/call_bag.py [hunk 2]
```diff
@@ -313,123 +312,33 @@ def pr_curve_old(self, pep_iz_subset=None, n_steps=50):
...
         scores = self.df["scores"].values
 
+        #FIXME: determine if this is a true statement, fix if not
         # At this point true, pred, scores are sorted WHOLE SET OF ALL PEPTIDES
```
Comment (zack-erisyon) [inline plaster/run/call_bag.py:338]: Just change word "peptide" to protein

File: plaster/run/call_bag.py [hunk 3]
```diff
@@ -313,123 +312,33 @@ def pr_curve_old(self, pep_iz_subset=None, n_steps=50):
...
 
-        # If a subset is not request then assume ALL are wanted
+        #FIXME: what should 'pep_iz_subset' be named now?
+        # If a subset is not requested then assume ALL are wanted
         if pep_iz_subset is None:
             pep_iz_subset = np.unique(
                 np.concatenate((self.df.true_pep_iz[1:], self.df.pred_pep_iz))
```
Comment (zack-erisyon) [inline plaster/run/call_bag.py:344]: It should be pro_iz_subset now

File: plaster/run/call_bag.py [hunk 4]
```diff
@@ -440,9 +349,17 @@ def pr_curve_new(self, pep_iz_subset=None, n_steps=50):
...
+                                pred, \
+                                scores, \
+                                true_in_subset_mask, \
+                                pred_in_subset_mask, \
+                                pop_iz_subset=pep_iz_subset, \
+                                n_steps=n_steps)
 
+
+    def pr_curve_either(self,pop,true,pred,scores,true_in_subset_mask, pred_in_subset_mask, pop_iz_subset, n_steps=50):
         # At this point, true_ and pred_in_subset_mask are masks on the original set.
```
Comment (zack-erisyon) [inline plaster/run/call_bag.py:363]: Doc string please. What is "pop"?

File: plaster/run/call_bag.py [hunk 5]
```diff
@@ -523,21 +440,21 @@ def pr_curve_new(self, pep_iz_subset=None, n_steps=50):
...
         if self._sim_result is not None:
-            filtered_pep_recalls = np.full_like(
+            filtered_pop_recalls = np.full_like(
                 self._sim_result.train_pep_recalls, np.nan
             )
-            filtered_pep_recalls[pep_iz_subset] = self._sim_result.train_pep_recalls[
-                pep_iz_subset
+            filtered_pop_recalls[pop_iz_subset] = self._sim_result.train_pep_recalls[
+                pop_iz_subset
             ]
```
Comment (zack-erisyon) [inline plaster/run/call_bag.py:451]: be careful -- this is refering to train_pep_recalls in this pro funtion

**Review by zack-erisyon — CHANGES_REQUESTED** [review]

File: plaster/run/call_bag.py (hunk 4 above)
Comment (rossdavidh) [reply plaster/run/call_bag.py:363]: "peptide or protein", but I'll add a docstring

File: plaster/run/call_bag.py (hunk 5 above)
Comment (rossdavidh) [reply plaster/run/call_bag.py:451]: Thanks!

File: plaster/run/call_bag.py [hunk 6]
```diff
@@ -313,123 +312,33 @@ def pr_curve_old(self, pep_iz_subset=None, n_steps=50):
...
-
-        # SCALE-DOWN all recall
-        prsa[:, recall_column] *= mean_recall
-
-        # SKIP all initial rows where the recall is zero, these clutter up the graph
-        # The return may thus have fewer than n_steps rows.
-        first_non_zero_i = utils.np_arg_first_where(prsa[:, recall_column] > 0.0)
-
-        filtered_prsa = prsa[first_non_zero_i:]
+        return self.pr_curve_either('pep', \
```
[the remark this answers is not in the document]
Comment (rossdavidh) [reply plaster/run/call_bag.py:315]: Hmmm...I somehow had the impression that "black" was run automatically when I did a git commit.

[the remark this answers is not in the document]
Comment (ossareh) [reply plaster/run/call_bag.py:315]: AFAICT it's only setup in erisyon/internal.

The internal repository has [`.pre-commit-config.yaml`](https://github.com/erisyon/internal/blob/master/.pre-commit-config.yaml), which is required for the `git commit` portion of this. However, in the container, it has recently been disabled by @zack ( https://github.com/erisyon/internal/blame/master/scripts/p_internal.py#L731-L733 ).

If you want to enable it for erisyon/plaster you'll need to create a config file, and then issue `pre-commit install`.

[the remark this answers is not in the document]
Comment (rossdavidh) [reply plaster/run/call_bag.py:315]: I see, thanks!

File: plaster/run/call_bag.py [hunk 7]
```diff
@@ -210,11 +210,13 @@ def average_classifier_scores_for_class(
         return n_calls, avg_scores
 
     def true_peps__pros(self):
+        self.df["Order"] = np.arange(len(self.df))
```
Comment (ossareh) [inline plaster/run/call_bag.py:213]: Is `"Order"` coming from something external? I"m surprised to see a capital "O" here.

**Review by ossareh — COMMENTED** [review]
I'm just a spectator here asking questions, so please don't index too highly on my thoughts :)

I see that this hasn't been blackened. Is it time to switch pre-commit on for the plaster repo?

File: plaster/run/call_bag.py (hunk 7 above)
Comment (rossdavidh) [reply plaster/run/call_bag.py:213]: Really, I would prefer to drop that column after I'm done using it.  I'll do that before the end of the day.

**Comment by rossdavidh** [comment]
Changed to address Mike's concern on the capitalized key, and Zack's various comments from before today.  Let me know if there's something else I need to do to mark it as "ready", as far as I know.

**Review by danwinkler — APPROVED** [review]
Looks good, make sure to run it through black. I didn't go too deep into making sure the logic is correct.

File: plaster/run/call_bag.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (rossdavidh) [reply plaster/run/call_bag.py]: Done.

File: plaster/run/call_bag.py (hunk 2 above)
Comment (rossdavidh) [reply plaster/run/call_bag.py]: Done.

**Comment by rossdavidh** [comment]
Did all changes requested by Zack,  Mike, and Dan, will defer Dan's idea of an Enum to when we add a third type of PR curve, so that more time can be spent on how exactly that should be designed.  I think it's a good idea, but opens a bit of a bigger topic on how to generalize across PR types, which probably will happen better when we actually are ready to introduce a third one.  Since the protein PR curve is needed ASAP, and I want to consider the more generalized PR curve more carefully, I'm just going with the current method for the moment.  I'll add a task to the backlog to look into this more thoroughly in the future.

**Pull request merged**